cognite_toolkit/cdf_tk/load.py

```diff
@@ -465,57 +497,63 @@ def delete(self, ids: Sequence[int], drop_data: bool) -> int:
                     f"  [bold yellow]WARNING:[/] Not deleting group {g.name} with sourceId {g.source_id} as it is used by the current service principal."
                 )
                 print("     If you want to delete this group, you must do it manually.")
-                if g.name not in ids:
+                if g.name not in ids_:
                     print(f"    [bold red]ERROR[/] You seem to have duplicate groups of name {g.name}.")
                 else:
-                    ids.remove(g.name)
-        found = [g.id for g in groups if g.name in ids]
+                    ids_.remove(g.name)
+        found = [g.id for g in groups if g.name in ids_ and g.id]
         self.client.iam.groups.delete(found)
         return len(found)
 
-    def create(self, items: Sequence[Group], drop: bool, filepath: Path) -> GroupList:
+    def create(self, items: GroupList, drop: bool, filepath: Path) -> GroupList:
         if self.target_scopes == "all":
             to_create = items
         elif self.target_scopes == "all_skipped_validation":
             raise ValueError("all_skipped_validation is not supported for group creation as scopes would be wrong.")
         elif self.target_scopes == "resource_scoped_only":
-            to_create = []
+            to_create = GroupList([])
             for item in items:
                 item.capabilities = [
-                    capability for capability in item.capabilities if type(capability.scope) in self.resource_scopes
+                    capability
+                    for capability in item.capabilities or []
+                    if type(capability.scope) in self.resource_scopes
                 ]
                 if item.capabilities:
                     to_create.append(item)
         elif self.target_scopes == "all_scoped_only" or self.target_scopes == "all_scoped_skipped_validation":
-            to_create = []
+            to_create = GroupList([])
             for item in items:
                 item.capabilities = [
-                    capability for capability in item.capabilities if type(capability.scope) not in self.resource_scopes
+                    capability
+                    for capability in item.capabilities or []
+                    if type(capability.scope) not in self.resource_scopes
                 ]
                 if item.capabilities:
                     to_create.append(item)
         else:
             raise ValueError(f"Invalid load value {self.target_scopes}")
 
         if len(to_create) == 0:
-            return []
+            return GroupList([])
         # We MUST retrieve all the old groups BEFORE we add the new, if not the new will be deleted
-        old_groups = self.client.iam.groups.list(all=True).data
-        created = self.client.iam.groups.create(to_create)
+        old_groups = self.client.iam.groups.list(all=True)
+        created = cast(GroupList, self.client.iam.groups.create(to_create))
```

Comment:
Why is a cast needed here?

Reply:
Bug in the SDK, it does not overload `.create` properly. I have fixed it in my PR.